accelerometer/circadianRhythms.py

```diff
@@ -0,0 +1,125 @@
+"""Module to support calculation of metrics of circadian rhythm from acc data"""
+
+import gzip
+import numpy as np
+import pandas as pd
+import pytz
+import sys
+import scipy as sp
+from scipy import fftpack
+from datetime import timedelta
+
+def calculatePSD(e, epochPeriod, fourierWithAcc, labels, summary):
+    """Calculate the power spectral density from fourier analysis of a 1 day frequency
+    
+    :param pandas.DataFrame e: Pandas dataframe of epoch data
+    :param int epochPeriod: Size of epoch time window (in seconds)
+    :param bool fourierWithAcc:True calculates fourier done with acceleration data instead of sleep data
+    :param list(str) labels: Activity state labels
+    :param dict summary: Output dictionary containing all summary metrics
+
+    :return: Write dict <summary> keys 'PSD-<W/Hz>'
+    """
+    if fourierWithAcc:
+        y = (e['accImputed'])
+    else:
+        cols = []
+        # get imputed variable names for each activity type
+        # NB - make sure sleep comes first
+        for accType in labels:
+            col = accType + 'Imputed'
+            cols += [col]
+      
+        idx = cols.index('sleepImputed') #gets index of sleep label
+        # collects the sleep column from data frame assumes sleep if it is the highest imputed value
+        y = (np.argmax(e[cols].values, axis=1) == idx).astype('int')*2-1
+        
+    n = len(y)
+    k = len(y)*epochPeriod/(60*60*24)
+    e = -2.j * np.pi * k * np.arange(n) / n
+    # finds the power spectral density for a one day cycle using fourier analysis 
+    res = np.sum(np.exp(e) * y, axis=-1)/n
+    PSD = np.abs(res)**2
+    summary['PSD'] = PSD
+
```

Comment:
+ 3 space gap

Reply:
fixed in ce9842906efbd024c24d668b2d3bd0b3f429f559